code/weapon/weapons.cpp

```diff
@@ -9083,7 +9165,7 @@ void weapon_render(object* obj, model_draw_list *scene)
 					scaled_head_radius,
 					&tailp,
 					scaled_tail_radius,
-					alpha, alpha, alpha);
+					bitmap_r, bitmap_g, bitmap_b);
```

Comment:
IIUC, this is because previously the laser bitmap (the main one, not the glow) could never have its color modified. It could be made transparent by various stuff, but unlike the glow bitmap, there was no way to recolor it in-engine, so it wasn't necessary to bother with the individual color values.